tests/ui/cypress/specs/multi_user_openid_connect.js

```diff
@@ -122,6 +152,14 @@ describe('OpenID Connect users', () => {
     // by the time Krill verifies it!
     cy.wait(6000)
 
+    // Try to create a CA, by typing in the input, clicking the 'Create CA' button
+    // and then clicking 'Ok'. This should fail, since the token can't be refereshed.
+    cy.intercept('POST', '/api/v1/cas').as('createCA')
+    cy.get('main input').type('some-handle-name')
+    cy.get('main button').click()
+    cy.get('.el-message-box__btns button:nth-child(2)').click()
+    cy.wait('@createCA').its('response.statusCode').should('eq', 401)
```

Comment:
I was wondering from reading the test what the UI experience is for the user, and that it would be good to verify that here. However, on trying it myself I find a problem. If I login as user `shorttokenwithoutrefresh@krill` then I receive a short-lived access token and no refresh token. When the create CA form is submitted after the token has been allowed to expire I therefore see an error in the UI:

> A (temporary) error occured while trying to authenticate your request. Please try again later.

Unfortunately this guidance is incorrect. No amount of retrying will help as the token is invalid and cannot be refreshed and so remains invalid. In this case we need to direct the user back to the login form. We should perhaps even do so transparently and automatically in case the provider is able to use user-agent cookies to magically "extend" the session by automatically issuing a new access token.

Reply:
I didn't want to convey to the user that they might get away with trying the button again, but that they may also have to start all over again, that's why I said 'try again later' instead of 'retry'. That might not be clear enough, we may rephrase to something like: 'please try again later of try to log in again'?

This is the thing with transient errors, right? They may be transient, but they may also be take way too long to resolve

Reply:
Maybe in this particular case we may also throw a ApiAuthPermanentError instead of ApiAuthTransientError

Reply:
In this particular case however no amount of waiting or retrying will ever work, only logging in again will work.

Reply:
> Maybe in this particular case we may also throw a ApiAuthPermanentError instead of ApiAuthTransientError

No, I don't think that's right either as the issue is not permanent. What needs to happen is that the user needs to re-login. Is it okay on HTTP 401 to redirect the user to the login process with a return-here-url-param to continue where they were before, perhaps transparently which is nice, or in the case that it isn't transparent because a login form is displayed or a provider error occurs, is it not nice because the user will think hang on why am I looking at a login form or an error from the provider? The middle ground is to include a login link in the error message. One way to handle this could be to cause the Login view to show as that has the ability to show only an error message with a link to the login form, the only downside being it takes the user away from the current thing they are seeing. Another alternative is to literally include a login link in the error message that is shown in the context of what the user is doing right now.

Reply:
I think what we can do in this PR is update this test to verify that the message is shown thereby highlighting the current behaviour, and leave a TODO comment at that point in the test script saying that we should in some way automatically re-login the user or guide them to relogin.

Reply:
I've updated the test to verify the message shown to the user. I decided not to add a TODO, rather any error experience improvements should be handled as part of #419. Still to do:
- [ ] Add a translation string for this new `api-auth-session-expired` error label.

Reply:
I added `api-auth-session-expired` to Lagosta PR #87 and synced this PR with the modified Lagosta `new-api-error-strings` branch.